radix-transactions/src/builder/transaction_builder.rs

```diff
@@ -365,28 +411,38 @@ impl TransactionV2Builder {
         self
     }
 
-    pub fn create_intent_and_subintent_signatures(&mut self) -> &TransactionIntentV2 {
+    pub fn transaction_header(mut self, transaction_header: TransactionHeaderV2) -> Self {
+        self.transaction_header = Some(transaction_header);
+        self
+    }
+
+    pub fn create_intent_and_subintent_info(&mut self) -> &TransactionIntentV2 {
         if self.intent_and_subintent_signatures.is_none() {
-            let (instructions, blobs, child_hashes) = self
-                .manifest
-                .as_ref()
-                .expect("Manifest must be provided")
-                .clone()
-                .for_intent();
+            let manifest = mem::take(&mut self.manifest)
```

Comment:
Minor.
`Option` does have it's own `take()` - maybe it would be more readable here.
```
let manifest = self
    .manifest
    .take()
    .expect("Manifest must be provided before this action is performed");
```